File: Samples/CS/CodeSamples/DataSmSample.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Inetlab.SMPP;
using Inetlab.SMPP.Common;
using Inetlab.SMPP.Parameters;
using Inetlab.SMPP.PDU;

namespace CodeSamples
{
    public class DataSmSample
    {
        public async Task<DataSmResp> SendDataSm(SmppClient client, string text)
        {
            DataSm  dataSm = new DataSm();
            dataSm.SourceAddress = new SmeAddress("1111");
            dataSm.DestinationAddress = new SmeAddress("79171234567");
            dataSm.DataCoding = DataCodings.UCS2;

            byte[] data = client.EncodingMapper.GetMessageBytes(text, dataSm.DataCoding);

            dataSm.Parameters.Add(new MessagePayloadParameter(data));

            return await client.SubmitData(dataSm);
        }
    }
}

File: Samples/VB.NET/CodeSamples/DataSmSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Threading.Tasks
Imports Inetlab.SMPP
Imports Inetlab.SMPP.Common
Imports Inetlab.SMPP.Parameters
Imports Inetlab.SMPP.PDU

Namespace CodeSamples
	Public Class DataSmSample
		Public Async Function SendDataSm(ByVal client As SmppClient, ByVal text As String) As Task(Of DataSmResp)
			Dim dataSm As New DataSm()
			dataSm.SourceAddress = New SmeAddress("1111")
			dataSm.DestinationAddress = New SmeAddress("79171234567")
			dataSm.DataCoding = DataCodings.UCS2

			Dim data() As Byte = client.EncodingMapper.GetMessageBytes(text, dataSm.DataCoding)

			dataSm.Parameters.Add(New MessagePayloadParameter(data))

			Return Await client.SubmitData(dataSm)
		End Function
	End Class
End Namespace
